clippy_lints/src/vec.rs

```diff
@@ -59,6 +59,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
             then {
                 // report the error around the `vec!` not inside `<std macros>:`
                 let span = arg.span
+                    .ctxt()
+                    .outer()
+                    .expn_info()
+                    .map(|info| info.call_site)
+                    .expect("unable to get call_site")
```

Comment:
That we have to do the exact same macro extraction twice here seems weird. That kind of indicates a rustc problem?

Reply:
It's actually pretty expected if a new expansion further expands the result of another expansion.